src/Agent.Worker/Worker.cs

```diff
@@ -91,24 +42,12 @@ private void InitializeSecrets(JobRequestMessage message)
                 var jobMessage = JsonUtility.FromString<JobRequestMessage>(channelMessage.Body);
                 ArgUtil.NotNull(jobMessage, nameof(jobMessage));
 
-                // Set the default thread culture.
-                string culture;
-                ArgUtil.NotNull(jobMessage.Environment, nameof(jobMessage.Environment));
-                ArgUtil.NotNull(jobMessage.Environment.Variables, nameof(jobMessage.Environment.Variables));
-                if (!jobMessage.Environment.Variables.TryGetValue(Constants.Variables.System.Culture, out culture))
-                {
-                    culture = null;
-                }
-
-                ArgUtil.NotNullOrEmpty(culture, nameof(culture));
-                HostContext.SetDefaultCulture(culture);
-
-                //initialize secret masks
-                InitializeSecrets(jobMessage);
-
-                Trace.Verbose($"JobMessage: {channelMessage.Body}");
+                // Initialize the secret masker and set the thread culture.
+                InitializeSecretMasker(jobMessage);
+                SetCulture(jobMessage);
```

Comment:
This method already has so much going on, I moved the culture logic into a private method to get it out of the way for readability.